### PR TITLE
update prettier to 192, use prettierconfig.

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Updates prettier to 1.9.2
- [x] Uses .prettierrc/.prettierignore so we can keep config in one place
- [x] Uses overrides in .prettierrc so we don't need per-file overrides in scripts
- [x] Can use opaque types in devices types now

Note: Make sure your IDE hard coded prettier settings are removed and that it picks up the .prettierrc

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -56,15 +56,7 @@
         "jsx-quotes": "off",
         "no-duplicate-imports": "off",
         "no-mixed-operators": "off",
-        "prettier/prettier": ["error", {
-          "singleQuote": true,
```
Comment (chrisnojima) [inline .eslintrc]: this is now in a prettierconfig file

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1 @@
+shared/common-adapters/icon.constants.js
```
Comment (chrisnojima) [inline .prettierignore]: this doesn't work due to a bug in prettier eslint: https://github.com/prettier/prettier-eslint/issues/67

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "bracketSpacing": false,
+  "printWidth": 110,
+  "semi": false,
+  "singleQuote": true,
+  "tabWidth": 2,
+  "trailingComma": "es5",
+  "overrides": [
```
Comment (chrisnojima) [inline .prettierrc]: woot, supports overrides so we can remove our own hand rolled overrides stuff

File: protocol/bin/flow.js [hunk 4]
```diff
@@ -8,14 +8,6 @@ const path = require('path')
 const camelcase = require('camelcase')
 const colors = require('colors')
 
-// load prettier rules from eslintrc
-const prettierOptions = json5.parse(
-  fs.readFileSync(path.join(__dirname, '../../.eslintrc'), {encoding: 'utf8'})
-).rules['prettier/prettier'][1]
-
-// Allow extra wide
-prettierOptions.printWidth = 9999
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: 🔥

File: protocol/bin/flow.js [hunk 5]
```diff
@@ -370,9 +362,8 @@ import type {Boolean, Bool, Bytes, Double, Int, Int64, Long, String, Uint, Uint6
     Object.keys(project.incomingMaps).map(im => `  '${im}'?: ${project.incomingMaps[im]}`).join(',') +
     '|}\n'
   const toWrite = [typePrelude, typeDefs.join('\n'), incomingMap].join('\n')
-  // const formatted = prettier.format(toWrite, prettierOptions)
-  // TODO disabling prettier short term
-  fs.writeFileSync(project.out, /*formatted*/ toWrite)
+  const formatted = prettier.format(toWrite, prettier.resolveConfig.sync(project.out))
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: @patrickxb can you verify this branch works. i think we have to remove this cause you couldn't run prettier effectively

File: shared/desktop/yarn-helper/action-creator-creator.js [hunk 6]
```diff
@@ -108,23 +114,15 @@ function compileReduxTypeConstant(ns: ActionNS, actionName: ActionName, desc: Ac
...
 }
 
-// load prettier rules from eslintrc
-const prettierOptions = json5.parse(
-  fs.readFileSync(path.join(__dirname, '../../../.eslintrc'), {encoding: 'utf8'})
-).rules['prettier/prettier'][1]
-
-// Allow extra wide
-prettierOptions.printWidth = 500
-
```
Comment (chrisnojima) [inline shared/desktop/yarn-helper/action-creator-creator.js]: 🔥

File: shared/desktop/yarn-helper/font.js [hunk 7]
```diff
@@ -266,12 +266,9 @@ ${/* eslint-disable */
 export type IconType = $Keys<typeof iconMeta_>
 export const iconMeta: {[key: IconType]: IconMeta} = iconMeta_
 `
-  // load prettier rules from eslintrc
-  const prettierOptions = json5.parse(
```
Comment (chrisnojima) [inline shared/desktop/yarn-helper/font.js]: 🔥

File: .eslintignore [hunk 8]
```diff
@@ -18,6 +18,5 @@
 /shared/markdown/parser.js
 /shared/node_modules
 /shared/test/setupTestFrameworkScriptFile.js
-/shared/constants/types/devices.js
```
Comment (chrisnojima) [inline .eslintignore]: woot, can use opaque types now

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: protocol/bin/flow.js (hunk 5 above)
Comment (patrickxb) [reply protocol/bin/flow.js]: works, thanks

**Pull request merged**